### ISSUE TITLE
Inline message accessibility role improvement

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#1645]
### Provide a general summary of the documentation request here

In the inline message accessibility tab roles `alert` and `status` are presented. I like the fact, that you make clear on the elements "Usage" page, that critical errors should be presented as a more disruptive element, such as a dialog.

I would add general information on using the variants : `warning`, `danger`, `info`, and `success`. The `alert` role is potentially very disruptive to assistive technology users and should be used in cases, where the information requires users immediate attention. Critical or time sensitive information (e.g., form errors when the user was filling a form, logout session timer is running out) should only be communicated using this role.

This might just be me foreshadowing a potential problem, but I've seen `role="alert"` used wrong in so many ways, that I would rather be extra careful than not.

### 👤 Your name

Valtteri Järvinen

### 🧢 Your Product/Team

External
### Tasks
- [x] add general info for variants, make documentation in general more clear
- [x] improve layout so information is easier to find (scannable)
- [x] we explain when you should use something but not the don'ts, this can be improved
- [ ] role=alert should only be used when variant is a danger message (and not for warning messages)

**Comment by anna-lach** [comment]
According to the conversation on the accessibility slack channel: https://sanoma.slack.com/archives/CFRULMYLD/p1747122952057719,
we remove `role` from the `inline-message` component. There will be `announcer` (`aria-live`) used instead of it.
